### PR TITLE
NFTs: filter instead of throw error when NFT has invalid network

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-####

## What changed (plus any additional context for devs)
* see original bug: https://rainbowhaus.slack.com/archives/C02C2FVC6N6/p1710811991043379
* happened bc the blast network was added to simplehash nft fetching, however it was not added to our util that converts simplehash chains to rainbow networks
* fixed by https://github.com/rainbow-me/rainbow/commit/25c1549f0b2667fc7e7a2e6ee29611335782b5a1
* in this PR: 
  * remove util that converts simplehash chains to rainbow networks, replace with retrieval directly from rainbow network config
  * unrelated cleanup to avoid type assertions

### CODE REVIEW

File: src/resources/nfts/simplehash/index.ts [hunk 1]
```diff
@@ -65,7 +65,12 @@ export async function fetchSimpleHashNFTListing(
   // array of all eth listings on OpenSea for this token
   let listings: SimpleHashListing[] = [];
   let cursor = START_CURSOR;
-  const chain = getSimpleHashChainFromNetwork(network);
+  const chain = RainbowNetworks.find(n => n.value === network)?.nfts?.simplehashNetwork;
```
Comment (skylarbarrera) [inline src/resources/nfts/simplehash/index.ts:68]: i think this only exists for zora and mainnet

Comment (skylarbarrera) [reply src/resources/nfts/simplehash/index.ts:68]: prob just need to map by simplehashNetwork || value or smthn 🤔

Comment (benisgold) [reply src/resources/nfts/simplehash/index.ts:68]: oh youre right. any harm in adding them for all networks?

Comment (skylarbarrera) [reply src/resources/nfts/simplehash/index.ts:68]: nah im coo with it just make the type required

File: src/resources/nfts/simplehash/index.ts [hunk 2]
```diff
@@ -41,8 +42,9 @@ export async function fetchSimpleHashNFTs(
   cursor: string = START_CURSOR
 ): Promise<{ data: SimpleHashNFT[]; nextCursor: string | null }> {
   const chainsParam = RainbowNetworks.filter(network => network.features.nfts)
-    .map(network => network.nfts?.simplehashNetwork || network.value)
+    .map(network => network.nfts.simplehashNetwork || network.value)
     .join(',');
+  console.log(chainsParam);
```
Comment (skylarbarrera) [inline src/resources/nfts/simplehash/index.ts:47]: log

File: src/resources/nfts/simplehash/index.ts [hunk 3]
```diff
@@ -41,8 +42,9 @@ export async function fetchSimpleHashNFTs(
   cursor: string = START_CURSOR
 ): Promise<{ data: SimpleHashNFT[]; nextCursor: string | null }> {
   const chainsParam = RainbowNetworks.filter(network => network.features.nfts)
-    .map(network => network.nfts?.simplehashNetwork || network.value)
+    .map(network => network.nfts.simplehashNetwork || network.value)
```
Comment (skylarbarrera) [inline src/resources/nfts/simplehash/index.ts:45]: dont need the fallback anymore yeah? the null type may end up causing issues here tho  🤔

**Review by skylarbarrera — APPROVED** [review]
lgtm

File: src/resources/nfts/simplehash/index.ts (hunk 3 above)
Comment (benisgold) [reply src/resources/nfts/simplehash/index.ts:45]: trueeee

**Pull request merged**